src/storage/entry_conversion.py

```diff
@@ -0,0 +1,135 @@
+from datetime import datetime
+from time import time
+from typing import List, Optional, Set
+
+from helperFunctions.data_conversion import convert_time_to_str
+from objects.file import FileObject
+from objects.firmware import Firmware
+from storage.schema import AnalysisEntry, FileObjectEntry, FirmwareEntry
+from storage.tags import collect_analysis_tags
+
+
+def firmware_from_entry(fw_entry: FirmwareEntry, analysis_filter: Optional[List[str]] = None) -> Firmware:
+    firmware = Firmware()
+    _populate_fo_data(fw_entry.root_object, firmware, analysis_filter)
+    firmware.device_name = fw_entry.device_name
+    firmware.device_class = fw_entry.device_class
+    firmware.release_date = convert_time_to_str(fw_entry.release_date)
+    firmware.vendor = fw_entry.vendor
+    firmware.version = fw_entry.version
+    firmware.part = fw_entry.device_part
+    firmware.tags = getattr(fw_entry, 'tags', {})
+    return firmware
+
+
+def file_object_from_entry(
+    fo_entry: FileObjectEntry,
+    analysis_filter: Optional[List[str]] = None,
+    included_files: Optional[Set[str]] = None,
+    parents: Optional[Set[str]] = None,
+) -> FileObject:
+    file_object = FileObject()
+    _populate_fo_data(fo_entry, file_object, analysis_filter, included_files, parents)
+    file_object.analysis_tags = collect_analysis_tags(file_object)
```

Comment:
weirdly, both functions actually seem to do mostly the same thing (one of them only returns tags with propagate which doesn't really make sense here, because this is not the function that collects tags from included files)
=> removed one of the functions